### PR TITLE
Prediction: tracking adaptation

### PR DESCRIPTION
**Pull request by HongyiSun**

### CODE REVIEW

File: modules/prediction/container/obstacles/obstacles_container.cc [hunk 1]
```diff
@@ -201,6 +283,33 @@ void ObstaclesContainer::BuildJunctionFeature() {
...
+    double obs_x = obstacle_ptr->latest_feature().position().x() + (timestamp_ -
+                   obstacle_ptr->latest_feature().timestamp()) *
+                   obstacle_ptr->latest_feature().raw_velocity().x();
+    double obs_y = obstacle_ptr->latest_feature().position().y() + (timestamp_ -
+                   obstacle_ptr->latest_feature().timestamp()) *
+                   obstacle_ptr->latest_feature().raw_velocity().y();
+    double dist = std::hypot(perception_obstacle.position().x() - obs_x,
+                             perception_obstacle.position().y() - obs_y);
+    if (dist < 3.0) {
+      return true;
```
Comment (YajiaZhang) [inline modules/prediction/container/obstacles/obstacles_container.cc]: Make 3.0 a gflag.

File: modules/prediction/container/obstacles/obstacles_container.cc [hunk 2]
```diff
@@ -160,6 +179,69 @@ void ObstaclesContainer::InsertFeatureProto(const Feature& feature) {
...
+          continue;
+        }
+        Obstacle* obstacle_ptr = obstacles_.GetSilently(obs_id);
+        if (obstacle_ptr == nullptr) {
+          AERROR << "Obstacle id [" << obs_id << "] with empty obstacle_ptr.";
+          break;
+        }
+        if (timestamp_ - obstacle_ptr->timestamp() > 0.5) {
+          ADEBUG << "Obstacle already reach time threshold.";
+          break;
```
Comment (YajiaZhang) [inline modules/prediction/container/obstacles/obstacles_container.cc]: Make 0.5 a gflag

File: modules/prediction/container/obstacles/obstacles_container.cc [hunk 3]
```diff
@@ -113,34 +121,45 @@ ObstaclesContainer::GetCurrentFramePredictableObstacleIds() const {
...
+      curr_frame_id_mapping_.end()) {
+    prediction_id = curr_frame_id_mapping_[prediction_id];
+    if (prediction_id != perception_obstacle.id()) {
+      ADEBUG << "Obstacle have got AdaptTracking, with perception_id: "
+             << perception_obstacle.id()
+             << ", and prediction_id: " << prediction_id;
+    }
+  }
+  if (prediction_id < -1) {
+    AERROR << "Invalid ID [" << prediction_id << "]";
```
Comment (YajiaZhang) [inline modules/prediction/container/obstacles/obstacles_container.cc]: Under which condition the prediction_id will be less than -1?

Comment (kechxu) [reply modules/prediction/container/obstacles/obstacles_container.cc]: I think in this case, < 0 may be okay

Comment (kechxu) [reply modules/prediction/container/obstacles/obstacles_container.cc]: -1 is the id of ego vehicle

**Pull request merged**